### DISCUSSION TITLE
Allow daemonized processes to rotate logs

### DISCUSSION THREAD

**Discussion by dd45e640b42e6da7da96faee3996ef7c** [filecoin-project/lotus#7083] [Storage Provider] [answered]
### Checklist

- [X] This is **not** a new feature or an enhancement to the Filecoin protocol. If it is,  please open an [FIP issue](https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0001.md).
- [X] This is **not** brainstorming ideas. If you have an idea you'd like to discuss, please open a new discussion on [the lotus forum](https://github.com/filecoin-project/lotus/discussions/categories/ideas) and select the category as `Ideas`.
- [X] I **have** a specific, actionable, and well motivated feature request to propose.

### Lotus component

lotus JSON-RPC API

### What is the motivation behind this feature request? Is your feature request related to a problem? Please describe.

log files get big fast especially when debug output is logged. to allow proper log "segmentation" and system hygiene a command to rotate the logfiles would be great!

### Describe the solution you'd like

lotus daemon log rotate --file=new.log

same for lotus-miner/worker

### Describe alternatives you've considered

none, the daemons keep the file handle open i assume so linux side logrotete solutions do not really work great

### Additional context

have a client sending over 10mb of log files to debug a problem, spend time on finding the relevant parts of it - with daily/weekly/whatever log files life would be easier

**Comment by lanzafame** [comment]
@f8-ptrk `logrotate` works fine with lotus logs. Underneath everything lotus uses Uber's zap logger which is designed to work with external log rotation tools.

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
i see that as working then. thanks @lanzafame

**Comment by lanzafame** [comment, marked as answer]
For anyone who comes across this at a later date, you will want to use the `copytruncate` option in your `logrotate` configuration for lotus/lotus-miner log files as this keeps the same file pointer and that way lotus/lotus-miner do not need to be restarted.
